### PR TITLE
More biomes

### LINKED ISSUE #1431: Feature Request: Mesa Biome
#### Issue description

Add Mesa Biome for DEFAULT level generator. some of my players request this. thank you.
#### Steps to reproduce the issue

nope
#### OS and versions

- Genisys: 1.2dev
- PHP: 7.0
- OS (server): -
- Game version: 0.15.2
#### Crashdump, backtrace, memory dumps, plugins list or other files

```

---
```

### PR DESCRIPTION
**Pull request by PeratX**
### Description

Implemented Beach and Mesa Biome

### Reason to modify

Close #1431

### CODE REVIEW

**Comment by NycuRO** [comment]
@Muqsit  not extend ocean ,lol

**Comment by NycuRO** [comment]
extend grassy biome

**Comment by KennanFatt** [comment]
Thanks @PeratX @dktapps

**Comment by dktapps** [comment]
@KennanFatt what? I had nothing to do with this, this is all credits to @PeratX as far as I know :P 
Will test this later, any seeds you would recommend?

**Comment by FaanMaario** [comment]
OMG IT'S AWESOME !

Tested, and wow amazing !

**Comment by Muqsit** [comment]
+1 ^-^

**Comment by Muqsit** [comment]
Some more biomes and plains is done lol.
Plains
Plains_M
And a sunflower plains biome..

**Comment by ZedCee** [comment]
@PeratX It's improved every time I see it.  Can't seem to find any Mesas, anyone have a seed for that?
- Deserts and Taiga are looking a bit better in size
- Plains are still big, but some variation on them like residuals (groups of trees,kind of like the old gen even with the raised land, and/or not with raised land), and variations like @Muqsit mentioned
- Cacti, beaches shouldn't have them, and there's just a lot of them in deserts
- Forests could sometimes use an oak and birch mixture, especially between the solid looking forests, the same goes for taiga
- Snowy forests might be nice

Again though, it's looking better every time I take a peek.  Keep up the good work! +1

**Comment by XFuryMCPE** [comment]
+1

**Comment by XFuryMCPE** [comment]
Nice!

**Comment by PeratX** [comment]
Because Mesa Biome needs temperature <= 2 (it is very rare!)

**Comment by ZedCee** [comment]
@PeratX I'll keep hunting

**Comment by ZedCee** [comment]
Tried out 420d9d1 so see the mesa.  Looking good so far! +1

File: src/pocketmine/level/generator/normal/biome/BeachBiome.php [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+		$sugarCane = new Sugarcane();
+		$sugarCane->setBaseAmount(0);
+		$sugarCane->setRandomAmount(2);
+		$this->addPopulator($sugarCane);
+
+		$this->setElevation(62, 65);
+	}
+
+	public function getName() : string{
```
Comment (mhsjlw) [inline src/pocketmine/level/generator/normal/biome/BeachBiome.php]: What in the ... type?

**Comment by Yoyu666** [comment]
lol @dktapps

**Pull request closed without merging**